Sc enable select

fqazi opened the pull request.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/81407)
<!-- Reviewable:end -->

fqazi committed:
sql/schemachanger: watch dog for connections

Previously, this workload had no watch dogs for connections,
to determine when a connection is hung. This was inadequate
because it made it hard to detect hangs within the workload.
To address this, this patch will add a connection watch
dog to detect if connection are making progress. If no
progress is detected, then a panic will be caused.

Release note: None

fqazi committed:
workload/schemachange: refactor op generation

To help support SELECT statements we are going to refactor,
the returned statements inside the schemachange operation
generator into structures.

Release note: None

fqazi committed:
workload/schemachange: add select statements

Previously, the schema change workload did not
select from any of the tables that were created.
This was inadequate because we didn't fully validate
readability of tables during schema changes. To address,
this patch adds supports for select statements.

Release note: None

fqazi committed:
sql/schemachange: skip constraint validation if column generation fails

Previously, we would still try to validate constraints, if any of
the generated columns could be evaluated. To address this, this
patch will skip constraint validation in these scenarios.
<what you did about it: To address this, this patch ...>

Release note: None

fqazi committed:
sql/schemachange: limit CTAS size

Previously, the schemachange workload could generate
pretty large tables because of joins on CTAS statements.
These could take a fairly long time to CREATE. To address,
this patch will limit the maximum number of rows in
these tables.

Release note: None
